sdk/core/Azure.Core.Experimental/tests/LowLevelClient/PetStoreClient.cs

```diff
@@ -90,6 +92,12 @@ public virtual async Task<Response> GetPetAsync(string id, RequestOptions option
                     }
                     else
                     {
+                        // NOTE: Can this become:
+                        // throw new RequestFailedException(message.Response);
+                        // ??
+                        // (that would be so much tidier!)
+                        // QUESTION: What happens in the Async case, do we still need it?
+                        // Because we can't have an async constructor.
```

Comment:
Open question here on what we would do in async APIs if we wanted to replace the _clientDiagnostics call with `new RequestFailedException(response)` instead.

Reply:
Per @pakrym:
It's not a huge problem 
most client calls are buffered
so all reads are sync
but for the last 1% I wonder if we should auto buffer error responses